core/src/OpenMPTarget/Kokkos_OpenMPTarget_Exec.hpp

```diff
@@ -1309,15 +1299,13 @@ KOKKOS_INLINE_FUNCTION void parallel_reduce(
     const Impl::TeamVectorRangeBoundariesStruct<
         iType, Impl::OpenMPTargetExecTeamMember>& loop_boundaries,
     const Lambda& lambda, ValueType& result) {
-  result = ValueType();
-
   ValueType* tmp_scratch =
-      (ValueType*)loop_boundaries.team.impl_reduce_scratch();
+      static_cast<ValueType*>(loop_boundaries.team.impl_reduce_scratch());
 #pragma omp barrier
   tmp_scratch[0] = ValueType();
 #pragma omp barrier
 
-#pragma omp simd reduction(+ : tmp_scratch[:1])
+#pragma omp for simd reduction(+ : tmp_scratch[:1])
```

Comment:
Please comment on this change

Reply:
Since `simd` does not do anything inside the target region, without `omp for` all threads will run all the iterations of the consecutive for-loop. This is needed only for the TeamVectorRange, for the ThreadVectorRange the `omp for` is introduced in the TeamThreadRange.